### DISCUSSION TITLE
Page.css overrides Layout.css

### DISCUSSION THREAD

**Discussion by edward1127** [vercel/next.js#51645] [Help]
### Summary

I'm using Next.js 13 with an app directory and Tailwind CSS. Occasionally, the badge color does not render correctly if the layout.css is overridden by the page.css. I'm curious about the possible root cause behind this inconsistent overriding behavior.
 
overridden by page.css:
![image](https://github.com/vercel/next.js/assets/45161984/6c1ffda9-abf9-4328-bc06-c2d719eae82d)
![image](https://github.com/vercel/next.js/assets/45161984/6096748c-c38b-4444-98fd-c55170a61faa)
layout.css is not overridden.
![image](https://github.com/vercel/next.js/assets/45161984/782cc6a3-3f65-465a-a557-af85d76b4aed)
![image](https://github.com/vercel/next.js/assets/45161984/20dde7f8-da5b-4d6e-8ca9-3b0e043b2571)

**Comment by icyJoseph** [comment]
Are you seeing this during dev mode? 

Global classnames like that, are bound to cause headaches. It all depends on which style tag is injected last... and in dev mode, I think as you make changes, they get re-inserted to the DOM.

**Reply by edward1127** [reply]
yep, I'm seeing this in dev mode. Do you know if there's a way to guarantee the insertion order?

**Comment by ironvale0** [comment]
I have the same problem, using css modules and route groups, styles for components used inside page and layout exists in both css bundles and in case when to component inside layout added some classes, they are overrided by page css. Looks like need to strip styles that are already exists inside modules for layouts below.
